CommonTools/PileupAlgos/plugins/PuppiProducer.cc

```diff
@@ -103,7 +123,33 @@ void PuppiProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
     int    pVtxId = -9999; 
     bool lFirst = true;
     const pat::PackedCandidate *lPack = dynamic_cast<const pat::PackedCandidate*>(&(*itPF));
-    if(lPack == 0 ) {
+    if (fUsePVAssignmentMap) {
+      const reco::PFCandidate *pPF = dynamic_cast<const reco::PFCandidate*>(&(*itPF));
```

Comment:
looks like the ```if (lPack``` should be placed back where it was and the cast inside ```if (fUsePVAssignmentMap)``` should be removed